Deprecation issued in Java source

The Scala compiler should not emit warnings in Java sources.

```java
public enum En {
  @Deprecated Um
}

public @interface Ann {
  En value();
}

public class A {
  @Ann(En.Um)
  public static void te() { return; }
}
```

gives

```
$> sc Test.java -Ypickle-java -deprecation
/Users/luc/code/scala/scala13/sandbox/Test.java:10: warning: Java enum Um in Java enum En is deprecated
  @Ann(En.Um)
          ^
1 warning
```

--- a/src/compiler/scala/tools/nsc/typechecker/Typers.scala
+++ b/src/compiler/scala/tools/nsc/typechecker/Typers.scala
@@ -1052,7 +1052,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       }
       def adaptConstant(value: Constant): Tree = {
         val sym = tree.symbol
-        if (sym != null && sym.isDeprecated)
+        if (sym != null && !context.unit.isJava && sym.isDeprecated)
           context.deprecationWarning(tree.pos, sym)
         tree match {
           case Literal(`value`) => tree

--- a/test/files/pos/t13017/A.java
+++ b/test/files/pos/t13017/A.java
@@ -1,0 +1,12 @@
+enum En {
+  @Deprecated Um
+}
+
+@interface Ann {
+  En value();
+}
+
+public class A {
+  @Ann(En.Um)
+  public static void te() { return; }
+}

--- a/test/files/pos/t13017/Test.scala
+++ b/test/files/pos/t13017/Test.scala
@@ -1,0 +1,5 @@
+//> using options -Werror -Ypickle-java -deprecation
+
+object Test {
+  def t = A.te()
+}

